### DISCUSSION TITLE
Running Serverless Workflow Example on Conductor

### DISCUSSION THREAD

**Discussion by taoj-action** [Netflix/conductor#2216] [Q&A]
I am looking at [serverless workflow examples](https://github.com/serverlessworkflow/specification/tree/main/examples).  I am not quite sure how to handle the following cases:

- [Event-Based Transitions Example](https://github.com/serverlessworkflow/specification/tree/main/examples#Event-Based-Transitions-Example): for injecting event into the middle of a workflow.  I guess the conductor could use workers for polling the event or some known status repo but it may not efficient.
- [Reusing Function And Event Definitions](https://github.com/serverlessworkflow/specification/tree/main/examples#reusing-function-and-event-definitions)
- Especially, [Book Lending](https://github.com/serverlessworkflow/specification/tree/main/examples#book-lending): if we need to lend millions of books per day for an online library, how can event be injected into the middle of a particular workflow execution?

Thanks a lot!

**Comment by aravindanr** [comment]
@taojwmware Can you elaborate on your use-case? Conductor does not support serverless workflow definition.

**Reply by taoj-action** [reply]
Thanks for taking a look. My question is how Conductor can solve similar mentioned questions from the serverless workflow examples.     We normally have a small number of workflow definitions (in DSL) but a very large number of workflow running instances for the same workflow definition. How can we effectively keep track of workflow-id in order to complete a wait task by an external event? Thanks!

**Reply by apanicker-nflx** [reply]
> How can we effectively keep track of workflow-id in order to complete a wait task by an external event?

Are you planning on consuming events outside of Conductor to complete tasks from an external service?

Another approach could be to curate the externalId (workflowId/taskRefName combination) as part of the payload that is sent to the system processing the request. This externalId could then be used as passthrough by this external system which would send it as part of its event payload which can be used by Conductor event handler to complete the wait task within your workflow. This removes the requirement of storing the workflowId outside of Conductor within your system.
